Sources/PackageGraph/ModulesGraph+Loading.swift

```diff
@@ -29,6 +29,7 @@ extension ModulesGraph {
         requiredDependencies: [PackageReference] = [],
         unsafeAllowedPackages: Set<PackageReference> = [],
         binaryArtifacts: [PackageIdentity: [String: BinaryArtifact]],
+        prebuilts: [PackageIdentity: [String: PrebuiltLibrary]],
```

Comment:
What is `String` here? Could this be a dedicated type to make it clearer what the key is referring to?

Reply:
It's the name of the product. It's a String at the source as well, which is eventually the package manifest. I can add a comment at lest.

Reply:
Added comment.

Reply:
Made Product Identifiable and used the ID.